http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -207,6 +208,24 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
   )(implicit sr: SamResource[R], ev: Ask[F, TraceId]): F[Unit] =
     samDao.deleteResource(samResource, creatorEmail, googleProject)
 
+  override def lookupOriginatingUserEmail[R](petOrUserInfo: UserInfo)(implicit ev: Ask[F, TraceId]): F[WorkbenchEmail] =
```

Comment:
This sam call also returns an `enabled` flag -- I think Leo should read that flag and 401 if the user is disabled (this is essentially replacing OpenDJ).

Maybe we could do some refactoring in this DAO -- instead of separate methods to get subject/email maybe we could have 1 method that calls Sam `/register/user` and returns a `case class SamUserInfo(email, subject, enabled)`, and callers can just use that object.

Reply:
Yeah I can look into refactoring this. What is `subject` in terms of user info?

Reply:
Subject is like the internal userid in Sam (e.g. primary key). Ex:
```
{
  "adminEnabled": true,
  "enabled": true,
  "userEmail": "ron.weasley@test.firecloud.org",
  "userSubjectId": "114763077412354570085"
}
```
Email is typically not a good identifier for a user since it can change.